sentry-android-core/src/main/java/io/sentry/android/core/AndroidTransactionProfiler.java

```diff
@@ -41,7 +43,7 @@
   private @Nullable File traceFile = null;
   private @Nullable File traceFilesDir = null;
   private @Nullable Future<?> scheduledFinish = null;
-  private volatile @Nullable ITransaction activeTransaction = null;
+  @VisibleForTesting volatile @Nullable ITransaction activeTransaction = null;
```

Comment:
Can we test the class without `VisibleForTesting`? I see this almost in every class.

Reply:
Avoiding it would make it harder/less readable to test the code without increasing the visibility of the field.
I realized that when using `org.jetbrains.annotations.VisibleForTesting` Android studio doesn't care whether you use it in tests or outside them, and builds in any case.  
However, using `androidx.annotation.VisibleForTesting` makes Android studio complain if you access that field outside tests.  
Of course, that would work in sentry-android-* only. Would it make any difference?

Reply:
The problem is that we could not use `androidx.annotation` in non-Android projects and that would make 2 different packages of annotations across the project, so we preferred to keep the standard.

You can configure AS to warn in such cases well.

You can also test the behavior of the method without using `VisibleForTesting`, for example, when you finish the transaction 2 times, you can `verify` that the `captureTransaction` method was called only once, so the `activeTransaction` was actually null during the 2nd time, instead of asserting the field directly.

Usually `VisibleForTesting` is only necessary when you are working with static fields or when its technically not possible to test the behaviour, it does not like this use case for this field.